### PR TITLE
added animation to the stepper - made various properties IBInspectable

### PR DESCRIPTION
**Pull request by farshidce**

### CODE REVIEW

**Comment by Cee** [comment]
I'll check this pr tonight. :)

File: Class/PFStepper.swift [hunk 1]
```diff
@@ -8,47 +8,84 @@
...
-            
             let isInteger = floor(value) == value
-            
+            let animation = CATransition()
+            animation.timingFunction = CAMediaTimingFunction(name: kCAMediaTimingFunctionEaseInEaseOut)
+            animation.type = kCATransitionFade
+            animation.duration = 0.75
+            bottomButton.layer.add(animation, forKey: "kCATransitionFade");
+            topButton.layer.add(animation, forKey: "kCATransitionFade");
+
```
Comment (Cee) [inline Class/PFStepper.swift:24]: Remove `;`

File: Class/PFStepper.swift [hunk 2]
```diff
@@ -8,47 +8,84 @@
...
+    @IBInspectable open var maximumValue: Double = 24 {
+        didSet {
+            if maximumValue < minimumValue {
+                minimumValue = maximumValue
+            }
+            value = min(maximumValue, minimumValue)
+            initButtonValues()
+        }
+    }
+    @IBInspectable open var stepValue: Double = 1;
```
Comment (Cee) [inline Class/PFStepper.swift]: Remove `;`

File: Class/PFStepper.swift [hunk 3]
```diff
@@ -8,47 +8,84 @@
...
+    @IBInspectable open var autorepeat: Bool = true
+    @IBInspectable open var showIntegerIfDoubleIsInteger: Bool = true
+    fileprivate var topButtonText: String = ""
+    fileprivate var bottomButtonText: String = "";
```
Comment (Cee) [inline Class/PFStepper.swift]: Remove `;`

File: Class/PFStepper.swift [hunk 4]
```diff
@@ -74,11 +111,34 @@ open class PFStepper: UIControl {
...
+
+    func createBottomUpAnimateButton(_ value: Int) -> UIButton {
+        let button = UIButton()
+        button.setTitle(String(stringInterpolationSegment: Int(value)), for: UIControlState())
+        button.setTitleColor(self.buttonsTextColor, for: UIControlState())
+        button.backgroundColor = self.backgroundColor
+        button.titleLabel?.font = self.buttonsFont
+        button.frame = CGRect(x: 0, y: bounds.size.height / 2, width: bounds.size.width,
+                height: bounds.size.height / 2)
+        return button;
```
Comment (Cee) [inline Class/PFStepper.swift]: Remove `;`

File: Class/PFStepper.swift [hunk 5]
```diff
@@ -74,11 +111,34 @@ open class PFStepper: UIControl {
...
+    func createUpDownAnimateButton(_ value: Int) -> UIButton {
+        let button = UIButton()
+        button.setTitle(String(stringInterpolationSegment: Int(value)), for: UIControlState())
+        button.setTitleColor(self.buttonsTextColor, for: UIControlState())
+        button.backgroundColor = self.buttonsBackgroundColor
+        button.titleLabel?.font = self.buttonsFont
+        button.alpha = 0.5
+        button.frame = CGRect(x: 0, y: 0, width: bounds.size.width,
+                height: bounds.size.height / 2)
+        return button;
```
Comment (Cee) [inline Class/PFStepper.swift]: Remove `;`

**Review by Cee — CHANGES_REQUESTED** [review]
1. `@IBDesignable` is good and easy to use. 👍 
2. Animation is optional in this class/stepper. We could choose whether animation is on/off.

So could you please split this pr into two different prs? One for animation (add `needsAnimation` property to this class as well as create an initializer), another one for `@IBDesignable`.

**Comment by farshidce** [comment]
closing this and creating a new one with the changes suggested

**Pull request closed without merging**